SystemManager/Src/SM_States.cpp

```diff
@@ -0,0 +1,243 @@
+
+    // Immediately move to Disarm mode
+    system_manager->setState(DisarmMode::getInstance());
+}
+
+SystemState& BootMode::getInstance() {
+    static BootMode singleton;
+    return singleton;
+}
+
+/********************
+ * Disarm Mode
+ ********************/
+void DisarmMode::execute(SystemManager *system_manager) {
+    // TODO Check if we received new PID tunings and call the AM function here
+
+    // Check all arming mechnisms triggered then move to GroundOp
+    // TODO
+
+    // Check switch here that changes modes
+    system_manager->rc_data = Los_Link::getInstance().getRx();
+    if (system_manager->rc_data.rx_channels[SM::RC_ARM_CHANNEL] > 90) {
+        system_manager->setState(GroundOpMode::getInstance());
+    }
+}
+
+SystemState& DisarmMode::getInstance() {
+    static DisarmMode singleton;
+    return singleton;
+}
+
+/********************
+ * Ground Op Mode
+ ********************/
+void GroundOpMode::execute(SystemManager *system_manager) {
+    // To move to flight check that armed switch is high,
+    // throttle and pitch are 0 (both sticks down)
+    system_manager->rc_data = Los_Link::getInstance().getRx();
+    if (system_manager->rc_data.rx_channels[SM::RC_ARM_CHANNEL] > 90 &&
+        system_manager->rc_data.rx_channels[SM::RC_THROTTLE_CHANNEL] == 0 &&
+        system_manager->rc_data.rx_channels[SM::RC_PITCH_CHANNEL]) {
+        system_manager->setState(FlightMode::getInstance());
+    }
+}
+
+void GroundOpMode::exit(SystemManager *system_manager) {
+    // Delete SM to AM queue to prevent PID tuning midair (not yet in Nov 27th ver)
+    // osMessageQueueReset(system_manager->SM_to_AM_queue);
+    // system_manager->SM_to_AM_queue = NULL;
+}
+
+SystemState& GroundOpMode::getInstance() {
+    static GroundOpMode singleton;
+    return singleton;
+}
+
+/********************
+ * Flight Mode
+ ********************/
+void FlightMode::enter(SystemManager *system_manager) {
+    // Start the PM thread and increase TM speed (have to start different task)
+    // xTaskCreate(system_manager->PMOperationTask, "PM Thread", &system_manager->PM_handle);
+    // vTaskDelete(system_manager->TM_handle);
+    // xTaskCreate(system_manager->TMOperationTask, "TM Thread", &system_manager->TM_handle);
+
+    // system_manager->SM_to_PM_queue = osMessageQueueNew(1, 256, NULL);
+    // system_manager->PM_to_AM_queue = osMessageQueueNew(1, 256, NULL);
+
+    // JUST for the Nov 27th deadline flight:
+
+    system_manager->SM_to_AM_queue = osMessageQueueNew(1U, 256U, NULL);
+    system_manager->attitude_manager.setSmQueue(system_manager->SM_to_AM_queue);
+    AM::AttitudeManager* am_pointer = &system_manager->attitude_manager;
+    xTaskCreate(system_manager->AMOperationTask, "AM Thread", 400U,
+                (void *)am_pointer, osPriorityNormal,
+                &system_manager->AM_handle);
+    // Task will automatically be added to the FreeRTOS scheduler
```

Comment:
Do we need to add a delay so that it arms and doesn't immediately try to pitch back?

Reply:
I made sure the throttle is set to 0, so am not too worried. It will be a clear sign the drone is armed.

Reply:
I guess I imagined the motors would be on in an "armed" state. I think we can leave this for now, and if it becomes an issue we can address this later.